siliconcompiler/schema/schema_obj.py

```diff
@@ -528,12 +531,18 @@ def _normalize_value(value, sc_type, error_msg, allowed_values):
         except TypeError:
             raise TypeError(error_msg) from None
 
-        if sc_type in ('str', 'file', 'dir'):
+        if sc_type == 'str':
             if isinstance(value, str):
                 return value
             else:
                 raise TypeError(error_msg)
 
+        if sc_type in ('file', 'dir'):
+            if isinstance(value, str) or isinstance(value, pathlib.Path):
```

Comment:
you can do `isinstance(value, (str, pathlib.Path))`

Reply:
Thanks, didn't know this was a thing but I like it! Changed over.